netplan/cli/commands/set.py

```diff
@@ -47,19 +48,57 @@ def run(self):
         self.parse_args()
         self.run_command()
 
+    def split_tree_by_hint(self, set_tree) -> (str, dict):
+        FALLBACK_HINT = '70-netplan-set'
+        network = set_tree.get('network', {})
+        # A mapping of 'origin-hint' -> YAML tree (one subtree per netdef)
+        subtrees = dict()
+        global_keys = ['renderer', 'version']
+        for devtype in network:
+            if devtype in global_keys:
+                continue  # special handling of global keys down below
+            for netdef in network.get(devtype, []):
+                hint = FALLBACK_HINT
+                filename = utils.netplan_get_filename_by_id(netdef, self.root_dir)
+                if filename:
+                    hint = os.path.basename(filename)[:-5]  # strip prefix and .yaml
+                netdef_tree = {'network': {devtype: {netdef: network.get(devtype).get(netdef)}}}
+                # Merge all netdef trees which are going to be written to the same file/hint
+                subtrees[hint] = self.merge(subtrees.get(hint, {}), netdef_tree)
+
+        # Merge global_keys into one of the available subtrees
+        # Write to same file (if only one hint/subtree is available)
+        # Write to FALLBACK_HINT if multiple hints/subtrees are available, as we do not know where it is supposed to go
+        if network.get('renderer') is not None or network.get('version') is not None:
```

Comment:
```suggestion
        if any(network.get(key) for key in global_keys):
```

Maybe it's a good idea to also have this done dynamically like this, since there might be more `global_keys` somewhere in the future.

Reply:
This is a nice improvement, which I'll happily take! This way we only need to adopt the `GLOBAL_KEYS` constant whenever a new top level entry is added.